From defd0936ddfede3dc4b35c6d5fcea0d7d8fe3317 Mon Sep 17 00:00:00 2001
From: Chad Versace <chadversary@chromium.org>
Date: Wed, 5 Sep 2018 12:04:37 -0700
Subject: [PATCH 1/2] CHROMIUM: Disable EGL_KHR_partial_update

Disable it because it fails the Android CTS.

BUG=b:112057679
TEST=The list of extensions `adb logcat | grep 'SurfaceFlinger: extensions'`
  does not contain EGL_KHR_partial_update.

TODO(chadversary): Drop this patch after we diagnose why
  EGL_KHR_partial_update fails.
Change-Id: I3128700fc65ad3d5c496acde402b40ef0f2a2b25
Reviewed-on: https://chromium-review.googlesource.com/1208311
Commit-Ready: Chad Versace <chadversary@chromium.org>
Tested-by: Chad Versace <chadversary@chromium.org>
Reviewed-by: Ilja H. Friedel <ihf@chromium.org>
Reviewed-by: Benjamin Gordon <bmgordon@chromium.org>

(cherry picked from commit 2921bfe22d43cbf8fd07d9574aa3c4572c954917)
Signed-off-by: Gurchetan Singh <gurchetansingh@chromium.org>
---
 src/egl/drivers/dri2/platform_android.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index 232064b9e87e..f689feedcf1d 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -1624,7 +1624,10 @@ dri2_initialize_android(_EGLDriver *drv, _EGLDisplay *disp)
     */
    if (disp->Extensions.ANDROID_native_fence_sync) {
       disp->Extensions.EXT_buffer_age = EGL_TRUE;
-#if ANDROID_API_LEVEL >= 23
+#if 0
+      /* TODO(chadversary): Re-enable EGL_KHR_partial_update after we diagnose
+       * why it fails the Android CTS on Chrome OS.
+       */
       disp->Extensions.KHR_partial_update = EGL_TRUE;
 #endif
    }
-- 
2.20.1

